scipy/_lib/_util.py

```diff
@@ -43,7 +43,8 @@ def _lazywhere(cond, arrays, f, fillvalue=None, f2=None):
         if f2 is not None:
             raise ValueError("Only one of (fillvalue, f2) can be given.")
 
-    arrays = np.broadcast_arrays(*arrays)
+    args = np.broadcast_arrays(cond,  *arrays)
+    cond, arrays = args[0],  args[1:]
```

Comment:
The fix suggested in https://github.com/scipy/scipy/pull/13370#pullrequestreview-564665361 looks a bit different. Probably I just don't understand all the details--the discussion in the issues seems a bit confusing.

@WarrenWeckesser does this look ok? 

Is a regression/unit test for this private function overkill?

Reply:
I guess there's some contention about desired behavior maybe based on: https://github.com/scipy/scipy/issues/13602#issuecomment-785785613

Reply:
Thanks @tylerjereddy. gh-13602 was already closed by gh-13607, and a unit test was added there. This PR is no longer needed to fix `crystalball.entropy`, but it could still improve `_lazywhere`.

> Is a regression/unit test for this private function overkill?

No, I think there needs to be some testing of `_lazywhere` before we merge this. Besides passing existing tests, it needs to show that `_lazywhere` works like `where` when the `cond` argument needs to be broadcasted with the rest, if that's it's purpose. 

I don't think we usually improve/generalize private functions unless there is an immediate need, but this seems useful and could prevent bugs in the future, so I would be interested in adding it if it is well tested.

Reply:
For information, I need _lazywhere to broadcast cond for [#13589], which I am still working on. That proposed change to argsreduce relies on distributions to broadcast the arguments as necessary, mostly automatically but in some cases explicitly.  Also it seems like "the right thing" (TM). But it would also work to wait until it is actually needed.

Reply:
Either way. If you'd like to add this here, just add tests. When this merges, you can merge master into your gh-13589 branch to use it. If you'd prefer to close this and make the change to `_lazywhere` there, that's fine too.